src/lib/y2storage/bcache_cset.rb

```diff
@@ -0,0 +1,70 @@
+# encoding: utf-8
+
+# Copyright (c) [2018] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2storage/storage_class_wrapper"
+
+module Y2Storage
+  # A Bcache caching set
+  #
+  # This is a wrapper for Storage::BcacheCset
+  class BcacheCset < Device
+    include Yast::I18n
+    wrap_class Storage::BcacheCset
+
+    # @!method blk_devices
+    #   @return [Array<BlkDevice>] returns a list of devices used as caching ones in this set
+    storage_forward :blk_devices, as: "BlkDevice"
+
+    # @!method blk_devices
+    #   @return [Array<BlkDevice>] returns a list of devices used as caching ones in this set
+    storage_forward :bcaches, as: "Bcache"
+
+    # @!method uuid
+    #   @return [String] returns an UUID of set
+    storage_forward :uuid
+
+    # @!method self.create(devicegraph)
+    #   @param devicegraph [Devicegraph]
+    #   @return [BcacheCset]
+    storage_class_forward :create, as: "Bcache"
+
+    # @!method self.all(devicegraph)
+    #   @param devicegraph [Devicegraph]
+    #   @return [Array<BcacheCset>] all the bcache_csets in the given devicegraph,
+    #     in no particular order
+    storage_class_forward :all, as: "BcacheCset"
+
+    def inspect
+      "<BcacheCset #{uuid} #{blk_devices.inspect}>"
+    end
+
+    def display_name
+      textdomain "storage"
+      format(_("Cache Set for %s"), bcaches.map(&:name).sort.join(", "))
```

Comment:
Shouldn't we cover the case in which bcaches is empty?

Reply:
BTW, taking into account where it will be used, I would go for something shorter (maybe using `#basename` of devices) and with less uppercases.

https://github.com/yast/yast-storage-ng/blob/master/src/lib/y2partitioner/actions/resize_blk_device.rb#L106
https://github.com/yast/yast-storage-ng/blob/master/src/lib/y2partitioner/actions/edit_blk_device.rb#L107

Reply:
How it should look in that case? just "Cache Set" ?

Reply:
That should be enough, IMHO

Reply:
hmm, but when I check its usage. I think it will be better to inform user what other devices will be affected. I will prepare something.